src/core/sock/sockinfo_tcp.cpp

```diff
@@ -428,6 +428,13 @@ void sockinfo_tcp::set_xlio_socket(const struct xlio_socket_attr *attr)
 void sockinfo_tcp::set_entity_context(entity_context *ctx)
 {
     m_entity_context = ctx;
+    // To reuse Ultra API path of TX completions
+    m_p_group = ctx;
```

Comment:
This breaks our other logics where we check for m_p_group and then for m_entity_context to decide if the socket is ultra api or entity context